components/script/dom/window.rs

```diff
@@ -356,6 +356,11 @@ impl Window {
         self.parent_info
     }
 
+    pub fn top_level_browsing_context_id(&self) -> TopLevelBrowsingContextId {
```

Comment:
This is only called on top level windows, right? So to get a failure early in case we make the wrong assumption about the window, maybe we should fail if it's not a top level window?

Maybe `get_top_level_browsing_context_if_top() -> Result<id>` (can't find a better name).

Reply:
I was under the impression that `window_proxy.top_level_browsing_context_id()` would return the top level id, even if the 'current' window is not top level, let's say in the context of an iframe. See https://github.com/servo/servo/blob/master/components/script/dom/windowproxy.rs#L64

Reply:
I was just thinking of this as a safety net *if* we only call top_level_browsing_context_id on top level windows. Don't we only call top_level_browsing_context_id() to send messages originating from top level windows?

Reply:
Where do we use that function now?

Reply:
Ok I see what you mean, actually I just now realized I could remove this method since we decided to only attach the top level id when constellation forwards the message. So now script only needs to check if it is the top level, using `window.is_top_level`, before sending the message to constellation. Previously it was constellation that would check this before sending the message to the embedder, for example see https://github.com/servo/servo/pull/20707/files#diff-55c92a6a5ba7654ce45fe6fc6c63740fL1156